RUFAS/routines/animal/animal_manager.py

```diff
@@ -1105,54 +1119,55 @@ def _calc_ration_at_interval(self, feed: Feed) -> None:
 
         Parameters
         ----------
-        feed
+        feed : Feed
             Instance of the Feed class
+        pen : Pen
+            Instance of Pen class.
 
         """
         available_feeds = ration_driver.AvailableFeeds()
         available_feeds.feed_nutrients(feed)
-        for pen in self.all_pens:
-            if pen.is_populated:
-                pen.subset_class_feeds(feed)
-                pen_specific_feed_data = available_feeds.get_feed_data_from_feed_ids(pen.allocated_feeds)
-
-                ration_per_animal = {}
-                ration_vals = {}
-
-                while "status" not in ration_per_animal or ration_per_animal["status"].lower() != "optimal":
-                    if pen.animal_combination == AnimalCombination.CALF:
-                        ration_per_animal = CalfRationManager.optimize()
-                        ration_vals = {"ME_total": 0}
-                    else:
-                        ration_per_animal, ration_vals = RationManager.formulate_ration(
-                            pen, pen_specific_feed_data, self.ANIMAL_GROUPING_SCENARIO
-                        )
-
-                # recording ration nutrition information in pen
-                nutrient_amount, nutrient_conc = RationReporter.report_ration(ration_per_animal, feed.available_feeds)
-                pen.ration_nutrient_amount = nutrient_amount
-                pen.ration_nutrient_conc = nutrient_conc
-                pen.MEdiet = ration_vals["ME_total"]
-                pen.dry_matter_intake = nutrient_amount["dm"]
-
-                ration_report = {}
-                ration_report["nutrient_amount"] = nutrient_amount
-                ration_report["nutrient_conc"] = nutrient_conc
-
-                for animal in list(pen.animals_in_pen.values()):
-                    animal.set_ration(ration_per_animal, nutrient_amount["dm"])
-                    animal.set_p_intake(nutrient_amount["phosphorus"], nutrient_conc["phosphorus"])
-
-                ration_per_pen = {}
-                num_animals = len(pen.animals_in_pen)
-                for key in ration_per_animal:
-                    if key == "status":
-                        ration_per_pen[key] = ration_per_animal[key]
-                    else:  # feeds and price
-                        ration_per_pen[key] = ration_per_animal[key] * num_animals
-
-                pen.ration = ration_per_pen
-                pen.ration_per_animal = ration_per_animal  # Important
+        if pen.is_populated:
+            pen.subset_class_feeds(feed)
+            pen_specific_feed_data = available_feeds.get_feed_data_from_feed_ids(pen.allocated_feeds)
```

Comment:
```suggestion
        if not pen.is_populated:
            return
        pen.subset_class_feeds(feed)
        pen_specific_feed_data = available_feeds.get_feed_data_from_feed_ids(pen.allocated_feeds)
        ...
```
Not something significant, but this would reduce the amount of indentation in the method.